src/Analysis/Ast/Impl/Analyzer/ModuleWalker.cs

```diff
@@ -240,53 +242,101 @@ public ModuleWalker(IServiceContainer services, IPythonModule module, PythonAst
-                        if (!IsStubBetterType(memberType, stubMemberType)) {
-                            continue;
+                // If type does not exist in module, but exists in stub, declare it unless it is an import.
+                // If types are the classes, merge members. Otherwise, replace type from one from the stub.
+                switch (sourceType) {
+                    case null:
+                        // Nothing in sources, but there is type in the stub. Declare it.
+                        if (v.Source == VariableSource.Declaration) {
+                            Eval.DeclareVariable(v.Name, v.Value, v.Source);
                         }
-
-                        // Get documentation from the current type, if any, since stubs
-                        // typically do not contain documentation while scraped code does.
-                        memberType?.TransferDocumentationAndLocation(stubMemberType);
-                        cls.AddMember(name, stubMember, overwrite: true);
-                    }
-                } else {
-                    // Re-declare variable with the data from the stub unless member is a module.
-                    // Modules members that are modules should remain as they are, i.e. os.path
-                    // should remain library with its own stub attached.
-                    if (!(stubType is IPythonModule)) {
-                        sourceType.TransferDocumentationAndLocation(stubType);
-                        // TODO: choose best type between the scrape and the stub. Stub probably should always win.
-                        var source = Eval.CurrentScope.Variables[v.Name]?.Source ?? VariableSource.Declaration;
-                        Eval.DeclareVariable(v.Name, v.Value, source);
-                    }
+                        break;
+
+                    case PythonClassType cls when Module.Equals(cls.DeclaringModule):
+                        // If class exists and belongs to this module, add or replace
+                        // its members with ones from the stub, preserving documentation.
+                        // Don't augment types that do not come from this module.
+                        // Do not replace __class__ since it has to match class type and we are not
+                        // replacing class type, we are only merging members.
+                        foreach (var name in stubType.GetMemberNames().Except(new[] { "__class__", "__base__", "__bases__", "__mro__", "mro" })) {
+                            var stubMember = stubType.GetMember(name);
+                            var member = cls.GetMember(name);
+
+                            var memberType = member?.GetPythonType();
+                            var stubMemberType = stubMember.GetPythonType();
+
+                            if (builtins.Equals(memberType?.DeclaringModule) || builtins.Equals(stubMemberType?.DeclaringModule)) {
```

Comment:
Most recent looks like 
```cs
                            if (sourceType.IsBuiltin || stubType.IsBuiltin) {
                                // If stub type does not have an immediate member such as __init__() and
                                // rather have it inherited from object, we do not want to use the inherited
                                // since current class may either have its own of inherits it from the object.
                                continue;
                            }
```
which also prevents pulling inherited `__init__` and others from stub